src/uct/cuda/cuda_ipc/cuda_ipc_cache.h

```diff
@@ -17,35 +17,45 @@
 
 
 typedef struct uct_cuda_ipc_cache        uct_cuda_ipc_cache_t;
-typedef struct uct_cuda_ipc_cache_region uct_cuda_ipc_cache_region_t;
 typedef struct uct_cuda_ipc_rem_memh     uct_cuda_ipc_rem_memh_t;
 
 
-struct uct_cuda_ipc_cache_region {
-    ucs_pgt_region_t        super;        /**< Base class - page table region */
-    ucs_list_link_t         list;         /**< List element */
-    uct_cuda_ipc_key_t      key;          /**< Remote memory key */
-    void                    *mapped_addr; /**< Local mapped address */
-    uint64_t                refcount;     /**< Track inflight ops before unmapping*/
-};
+/**
+ * Structure associated with rcache region. ipc_handle is returned along with
+ * the start of the mapping to check if VA recycling has occurred.
+ */
+typedef struct uct_cuda_ipc_rcache_region {
+    ucs_rcache_region_t     super;
+    CUipcMemHandle          ipc_handle;
+    ucs_rcache_t            *rcache;
+    void                    *rem_base_address;
+    size_t                  rem_alloc_length;
+    void                    *mapping_start;
+} uct_cuda_ipc_rcache_region_t;
 
 
 struct uct_cuda_ipc_cache {
-    pthread_rwlock_t      lock;       /**< protests the page table */
-    ucs_pgtable_t         pgtable;    /**< Page table to hold the regions */
+    ucs_rcache_t          *rcache;    /**< rcache to hold local mappings of remote memory */
```

Comment:
imo, `uct_cuda_ipc_cache` is not needed anymore, rcache can be stored in md struct (also the name is a part of rcache struct, no need to keep it in `uct_cuda_ipc_cache`)